### PR TITLE
Added Compose screenshot generation support.

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `UNRELEASED.md`
+ [ ] `README.md`
+ [ ] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt [hunk 1]
```diff
@@ -34,86 +34,92 @@ import org.threeten.bp.LocalDate
...
-  ).map { it.toArgs() }
+    ViewScreenshot("Toast", "default", ::setupToast)
+  )
 
   init {
     CalendarDispatchers.setBackground(TestCoroutineDispatcher())
   }
 }
 
-data class Screenshot(
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt:110]: We only used this class to map it to array of arguments for GenerateScreenshots. It's been replaced with 2 corresponding functions

File: app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt [hunk 2]
```diff
@@ -34,86 +34,92 @@ import org.threeten.bp.LocalDate
...
-  val setup: (() -> Unit)? = null
-) {
-  fun toArgs() = arrayOf(name, screenshotName, setup)
-}
+fun ComposeScreenshot(
+  name: String,
+  screenshotName: String,
+  setup: (() -> Unit)? = null,
+): Array<Any?> =
+  arrayOf(name, screenshotName, "compose", setup)
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt:115]: We can change the path when we make the decision regarding Compose docs structure.

File: app/src/androidTest/java/net/skyscanner/backpack/docs/RemoteScreenGrab.kt [hunk 3]
```diff
@@ -27,8 +27,8 @@ import java.net.URL
 class RemoteScreenGrab(
   private val serverIp: String
 ) {
-  fun takeScreenshot(name: String) {
-    val url = URL("http://$serverIp:8888?name=$name")
+  fun takeScreenshot(name: String, path: String) {
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/docs/RemoteScreenGrab.kt:30]: Path from model is being passed to Node server

File: scripts/screenshot-server.js [hunk 4]
```diff
@@ -51,7 +51,7 @@ http
 
     const [componentName, ...screenShotName] = query.name.split('_');
 
-    const folderPath = path.join('docs', componentName, 'screenshots');
+    const folderPath = path.join(`${query.path}`, componentName, 'screenshots');
```
Comment (bvitaliyg) [inline scripts/screenshot-server.js:54]: And we use it to locate the output file

File: app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt [hunk 5]
```diff
@@ -34,86 +34,92 @@ import org.threeten.bp.LocalDate
...
+    ViewScreenshot("Text - View - Emphasized", "emphasized"),
+    ViewScreenshot("Text - View - Heavy", "heavy"),
+    ViewScreenshot("Text Field - Default", "default"),
+    ViewScreenshot("Text Field - With labels", "labels"),
+    ViewScreenshot("Text Spans", "default"),
+    ViewScreenshot("Spinner - Default", "default"),
+    ViewScreenshot("Spinner - Small", "small"),
     // Leave toast last as it stays visible in the screen for a while
-    Screenshot("Toast", "default", ::setupToast)
-  ).map { it.toArgs() }
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt:103]: No need to map anymore

File: app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt [hunk 6]
```diff
@@ -34,86 +34,92 @@ import org.threeten.bp.LocalDate
...
-    Screenshot("Switch", "default"),
-    Screenshot("Text - View - Default", "default"),
-    Screenshot("Text - View - Emphasized", "emphasized"),
-    Screenshot("Text - View - Heavy", "heavy"),
-    Screenshot("Text Field - Default", "default"),
-    Screenshot("Text Field - With labels", "labels"),
-    Screenshot("Text Spans", "default"),
-    Screenshot("Spinner - Default", "default"),
-    Screenshot("Spinner - Small", "small"),
+    ViewScreenshot("Badge", "all"),
```
Comment (bvitaliyg) [inline app/src/androidTest/java/net/skyscanner/backpack/docs/DocsRegistry.kt:37]: Renamed for consistency

**Review by marianeum — APPROVED** [review]

**Pull request merged**